### PR TITLE
Some bug fixes

### PR DESCRIPTION
**Pull request by nagmat84**
This fixes some oversights in https://github.com/LycheeOrg/basicModal/pull/3 which became unveiled by https://github.com/LycheeOrg/Lychee-front/pull/322 and also adds some additional methods to a modal dialog:

 * If the form elements of a dialog are constructed dynamically after its initial creation, the result values of those elements were not part of the result object of the dialog. Now, the modal dialog provides a method "`cacheFormElements`".
 * The dialog provides methods to hide the action and cancle button dynamically. Previously, this was only possible by using CSS selectors from the client code which violated the principle of the separation of concerns.
 * The `close` method accepts a callback which is invoked after the dialog has been closed. Previously, we had some lurking bugs in the fronend by running code which had to be executed after the dialog closed directly after the `close` method which did not consider that `close` is an async method.

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -302,13 +330,22 @@ export const getValues = function() {
...
+				values[inputElement.name] = parseInt(inputElement.value, 10);
 				break;
 			case 'file':
 				values[inputElement.name] = inputElement.files;
 				break;
 			default:
-				values[inputElement.name] = inputElement.value;
+				switch (inputElement.inputMode) {
+					case "numeric":
+						values[inputElement.name] = parseInt(inputElement.value, 10);
```
Comment (ildyria) [inline src/scripts/main.js:341]:
```suggestion
						values[inputElement.name] = inputElement.valueAsNumber;
```

File: src/scripts/main.js [hunk 2]
```diff
@@ -302,13 +330,22 @@ export const getValues = function() {
...
+						break;
+					case "decimal":
+						values[inputElement.name] = parseFloat(inputElement.value);
```
Comment (ildyria) [inline src/scripts/main.js:344]:
```suggestion
						values[inputElement.name] = inputElement.valueAsNumber;
```

File: src/scripts/main.js [hunk 3]
```diff
@@ -302,13 +330,22 @@ export const getValues = function() {
 				break;
 			case 'number':
 			case 'range':
-				values[inputElement.name] = parseInt(inputElement.value);
+				values[inputElement.name] = parseInt(inputElement.value, 10);
```
Comment (ildyria) [inline src/scripts/main.js:333]:
```suggestion
				values[inputElement.name] = inputElement.valueAsNumber;
```

**Review by ildyria — APPROVED** [review]
Not tested, but makes use of proper JS types conversions.

File: src/scripts/main.js (hunk 1 above)
Comment (nagmat84) [reply src/scripts/main.js:341]: No, unfortunately not. I had this that way during my first attempt and it did not work as expected. Unfortunately, I have forgotten the exact problem, but it was something stupid and was related to the "magic" auto-conversion of `valueAsNumber`. `valueAsNumber` does not reliably return a numeric value but may also return a `Date` object, if `valueAsNumber` find the value to be convertible into a date.

File: src/scripts/main.js (hunk 2 above)
Comment (nagmat84) [reply src/scripts/main.js:344]: See previous answer.

File: src/scripts/main.js (hunk 3 above)
Comment (nagmat84) [reply src/scripts/main.js:333]: And again.

**Review by ildyria — APPROVED** [review]
LGTM.

**Pull request merged**